base_field_validator/models/ir_model.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    See __openerp__.py about license
+#
+##############################################################################
+
+import re
+from openerp.tools.translate import _
+from openerp import SUPERUSER_ID
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning
+from openerp.osv import orm
+
+
+class IrModel(models.Model):
+
+    _inherit = 'ir.model'
+    validator_line_ids = fields.One2many(
+        'ir.model.validator.line', 'model_id', 'Validators')
+
+    @api.model
+    def check_vals(self, vals, model_name):
+        validator_lines = self.env['ir.model.validator.line'].search([
+            ('model_id.model', '=', model_name),
+            ('field_id.name', 'in', vals.keys())])
+        for validator_line in validator_lines:
+            pattern = re.compile(validator_line.regex_id.regex)
+            if vals[validator_line.field_id.name]:
+                if not pattern.match(vals[validator_line.field_id.name]):
+                    raise Warning(
+                        _('Expression %s not valid for %s') % (
+                            validator_line.regex_id.regex,
+                            vals[validator_line.field_id.name]))
+        return True
```

Comment:
Any public method is available at the RPC level, and you need to return something to be compatible with `XML-RPC` (JSON-RPC doesn't care, `None` is an acceptable value).
I don't know if there is an interest to call this method with `XML-RPC`, but this is the main reason.